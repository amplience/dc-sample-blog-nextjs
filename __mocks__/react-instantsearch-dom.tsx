import React, { ReactElement } from 'react';

export const InstantSearch = (): ReactElement => <div>InstantSearch mock</div>;
