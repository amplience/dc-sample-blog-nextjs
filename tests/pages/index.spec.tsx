/* eslint-env jest */
import React from 'react';
import { findResultsState } from 'react-instantsearch-dom/server';
import Index from '../../pages/index';
import blogListFixture from '../fixtures/blog-list-one-blog.json';
import ShallowRenderer from 'react-test-renderer/shallow';

jest.mock('search-insights');
jest.mock('react-instantsearch-dom/server', () => ({
  findResultsState: jest.fn()
}));

jest.mock('react-instantsearch-dom', () => ({
  ...jest.requireActual('react-instantsearch-dom'),
  connectStateResults: templateFn => params => templateFn(params)
}));

jest.mock('algoliasearch', () => jest.fn().mockImplementation(() => ({ search: jest.fn() })));

describe('Index', () => {
  beforeEach(() => {
    process.env.ALGOLIA_APPLICATION_ID = 'algolia-app-id';
    process.env.SEARCH_API_KEY = 'algolia-search-key';
    process.env.SEARCH_INDEX_NAME_PRODUCTION = 'algolia-index-name';
    process.env.DYNAMIC_CONTENT_REFERENCE_ID = 'reference-id';
    process.env.GA_TRACKING_ID = 'ga-tracking-id';
    process.env.HITS_PER_PAGE = '9';
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  describe('components', () => {
    it('should render index', async () => {
      const component = ShallowRenderer.createRenderer();
      component.render(<Index {...blogListFixture} />);
      expect(component.getRenderOutput()).toMatchSnapshot();
    });
  });
});
