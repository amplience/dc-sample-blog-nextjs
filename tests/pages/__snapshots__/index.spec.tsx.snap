// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index components should render index 1`] = `
<_default>
  <b
    description="heading"
    title="Blog"
  />
  <InstantSearch
    createURL={[Function]}
    indexName="algolia-index-name"
    onSearchStateChange={[Function]}
    refresh={false}
    resultsState={null}
    searchClient={
      Object {
        "search": [MockFunction],
      }
    }
    searchState={Object {}}
    stalledSearchDelay={200}
  >
    <ConnectorWrapper
      clickAnalytics={true}
      hitsPerPage="9"
    />
    <ConnectorWrapper
      scrollOn="page"
    />
    <ConnectorWrapper
      scrollOn="menu"
    />
    <HeroBanner
      heading="heading"
    >
      <HeaderSearchBox
        placeholderText=""
      />
      <HeaderFacetBar />
    </HeroBanner>
    <Unknown />
    <SearchResultPagination />
  </InstantSearch>
  <JSXStyle
    id="2750196598"
  >
    footer{margin-top:120px;}
  </JSXStyle>
</_default>
`;
